### DISCUSSION TITLE
Is there a way to install the latest compatible version of a package automatically?

### DISCUSSION THREAD

**Discussion by nbro** [python-poetry/poetry#9198] [Q&A]
If I do `poetry add django`, I get an error that says that the latest version of Django is incompatible with Python 3.9. This is fine. Now, I don't want to look for the latest compatible version of a package in PyPI because this takes a lot of time. I want poetry to do this automatically for me, i.e. it should install the **latest compatible version of a package**, even if not the latest. 

**Is there a way to tell poetry not to fail if the latest version of a package is incompatible and instead try to install the latest compatible version?**

Maybe this should even be the default behaviour, but I don't want to argue about this. Definitely, this feature should exist, though.

**Comment by nbro** [comment]
Although this is not described in `poetry add --help`, we can do `poetry add "django@*"`. I found this suggestions [here](https://stackoverflow.com/a/69813861/3924118). However, the problem is that this will add the `*` to `pyproject.toml` as the version of `django`, which is clearly not what I want. I also don't want to first do `poetry add "django@*" --dry-run` and, once I've determined the latest compatible version, do `poetry add` again. There should be a command that does this automatically.

Could the help page of `poetry add` be updated with info about this topic anyway? I think there should be a more descriptive flag that needs to do this anyway.

**Reply by pblanis** [reply]
The workaround seems to no longer work with 2.1.1.
